ssg/build_remediations.py

```diff
@@ -280,40 +281,31 @@ def parse_from_file_with_jinja(self, env_yaml):
         if stripped_fix_text == "":
             return result
 
-        rule_platforms = set()
+        rule_specific_platforms = set()
+        inherited_platforms = set()
         if self.associated_rule:
             # There can be repeated inherited platforms and rule platforms
-            rule_platforms.update(self.associated_rule.inherited_platforms)
+            inherited_platforms.update(self.associated_rule.inherited_platforms)
             if self.associated_rule.platform is not None:
-                rule_platforms.update(self.associated_rule.platform)
-
-        platform_conditionals = []
-        for platform in rule_platforms:
-            if platform == "machine":
-                # Based on check installed_env_is_a_container
-                platform_conditionals.append('[ ! -f /.dockerenv ] && [ ! -f /run/.containerenv ]')
-            elif platform is not None:
-                # Assume any other platform is a Package CPE
-
-                # Some package names are different from the platform names
-                if platform in self.local_env_yaml["platform_package_overrides"]:
-                    platform = self.local_env_yaml["platform_package_overrides"].get(platform)
-
-                    # Workaround for plaforms that are not Package CPEs
-                    # Skip platforms that are not about packages installed
-                    # These should be handled in the remediation itself
-                    if not platform:
-                        continue
+                rule_specific_platforms = {p for p in self.associated_rule.platform if p not in inherited_platforms }
 
-                # Adjust package check command according to the pkg_manager
-                pkg_manager = self.local_env_yaml["pkg_manager"]
-                pkg_check_command = PKG_MANAGER_TO_PACKAGE_CHECK_COMMAND[pkg_manager]
-                platform_conditionals.append(pkg_check_command.format(platform))
+        inherited_conditionals = [self.generate_platform_conditional(p) for p in inherited_platforms]
+        rule_specific_conditionals = [self.generate_platform_conditional(p) for p in rule_specific_platforms]
+        # remove potential "None" from lists
+        inherited_conditionals = [p for p in inherited_conditionals if p != None]
+        rule_specific_conditionals = [p for p in rule_specific_conditionals if p != None]
 
-        if platform_conditionals:
+        if inherited_conditionals or rule_specific_conditionals:
             wrapped_fix_text = ["# Remediation is applicable only in certain platforms"]
 
-            all_conditions = " && ".join(platform_conditionals)
+            all_conditions = ""
+            if inherited_conditionals:
+                all_conditions += " && ".join(inherited_conditionals)
+            if rule_specific_conditionals:
+                if all_conditions:
+                    all_conditions += " && ( " + " || ".join(rule_specific_conditionals) + " )"
```

Comment:
I found a shellcheck warning, which I think we cannot easily get rid of.
It happens whenever there is an inherited platform and `machine` is one of the rule specific platforms.

```
In bash/rhel8-script-cis.sh line 326:                                                                                                        
if rpm --quiet -q gdm && ( [ ! -f /.dockerenv ] && [ ! -f /run/.containerenv ] ); then                                                       
                         ^-- SC2235: Use { ..; } instead of (..) to avoid subshell overhead.      
```
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/software/gnome/group.yml

ShellCheck reference: https://github.com/koalaman/shellcheck/wiki/SC2235